[ticket/15945] Allow redirection to subdomains

ajquick opened the pull request:
This change improves the way phpBB prevents redirection to outside
domains through the redirection function. This change allows phpBB to
redirect any domain or subdomain that is covered under the cookie
settings.

PHPBB3-15945

Checklist:

- [x] Correct branch: master for features
- [x] Tests pass
- [x] Code follows coding guidelines: 3.2.x
- [x] Commit follows commit message format

Tracker ticket :

https://tracker.phpbb.com/browse/PHPBB3-15945

ajquick commented:
And phpBB continues to throw a hissy fit over the commit messages. Anyone want to just make this change themselves?

DavidIQ commented:
This commit message is wrong: `Allow redirection to subdomains`

It should follow the guidelines here: https://area51.phpbb.com/docs/dev/3.2.x/development/git.html#commit-messages

ajquick commented:
Can't do anything about previous commit messages as phpBB looks at all commit messages. That's also not the issue of the first commit messages. It appears to be throwing a fit because I used Github's browser which uses Windows style line breaks instead of the expected Unix line breaks.

ajquick commented:
> Bug or minor improvements do not need a name, their branches should be called ticket/1234 with the correct ticket id.

DavidIQ commented:
That's completely unrelated to phpBB as we aren't looking at anything. Travis is looking at your commit messages that are in git. And, of course, doing things through the GitHub interface will cause unexpected results. Changing commit messages is also possible through the command line. Just need to issue the right git commands to do so.

ajquick commented:
I rebased and squashed the commits and also changed messages (yes through the command line!). That's why you see 6 commits that are all exactly the same. Travis looks at all the commits, rather than the one that changes the file. If one of them fails the whole thing fails. Can't revert and change it now it seems without starting over (a second time) it seems.

Unable to use Github when this is housed on Github? Seems a little counter productive don't you think?

DavidIQ commented:
Not really our fault that GitHub's interface does funny things to line endings.

Anyhow if you changed the commit message then it should be reflected in your commits after you do a force push to your branch so you're missing a step somewhere.

ajquick commented:
Finally got it.

ajquick commented:
At least. It should be getting it, but the testing is checking for CNET's CNAME and failing? Going to be pulling out my hairs.

DavidIQ commented:
That might be temporary and probably nothing to do with your changes. We'll have to check it out again tomorrow as it looks like a timeout.

ajquick commented:
Can anyone trigger another travis check for me on that script?

https://travis-ci.org/phpbb/phpbb/jobs/486216754

DavidIQ commented:
Running.

ajquick commented:
I guess _news.cnet.com_  just doesn't have a valid CNAME record anymore.

I have created a new pull request and bug related to this: https://github.com/phpbb/phpbb/pull/5527

ajquick commented:
Finally passing now that the CNAME bug is fixed.

hanakin commented:
@ajquick you seem to be making changes on a 3.2.x branch but you are pushing this pr against master which is a 4.0 branch. Please rebase this to 3.2.x or recreate on proper master.

ajquick commented:
> @ajquick you seem to be making changes on a 3.2.x branch but you are pushing this pr against master which is a 4.0 branch. Please rebase this to 3.2.x or recreate on proper master.

I thought all feature requests needed to go on Master? Would this be more of a bug fix in your opinion?

hanakin commented:
the second yes if you are submiting to master it needs to be based on master

marc1706 commented:
Closing this as explained in the tracker ticket: https://tracker.phpbb.com/browse/PHPBB3-15945?focusedCommentId=64280&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-64280
